src/trans/gpu/internal/ftdir_mod.F90

```diff
@@ -98,7 +98,7 @@ SUBROUTINE FTDIR(ALLOCATOR,HFTDIR,PREEL_REAL,PREEL_COMPLEX,KFIELD)
     PREEL_COMPLEX => PREEL_REAL
 #else
     CALL ASSIGN_PTR(PREEL_COMPLEX, GET_ALLOCATION(ALLOCATOR, HFTDIR%HREEL_COMPLEX),&
-      & 1_C_SIZE_T, INT(KFIELD*D%NLENGTF*SIZEOF(PREEL_COMPLEX(1)),KIND=C_SIZE_T))
+      & 1_C_SIZE_T, INT(KFIELD*D%NLENGTF*STORAGE_SIZE(PREEL_COMPLEX(1))/8,KIND=C_SIZE_T))
```

Comment:
```suggestion
      & 1_C_SIZE_T, 1_C_SIZE_T*KFIELD*D%NLENGTF*STORAGE_SIZE(PREEL_COMPLEX(1))/8)
```
It might be fine, not sure about the type of NLENGTF, but make sure you are not introducing new potential overflows. The value you compute is going to be 8X larger than before. Since you are changing this lines anyway, might be a good point to make those computations in JPIB or C_SIZE_T.

Reply:
note that the iso_c_binding function c_sizeof returns c_size_t and storage_size also accepts a KIND argument alternatively

Reply:
I think, what you are missing is my initial thought process, which was, that I thought that it already fixes the potental overflows (#162), which it does not, so I am fine with any solution. But: What I wanted to say initially is that the intermediate `KFIELD*D%NLENGTF*STORAGE_SIZE(...)` is 8 times larger than before, so a overflow is more likely than it was before, so we have to make sure that this expression does not overflow, by casing to C_SIZE_T early enough. But if you don't see overflows, you probably can just leave it like this and we can discuss it in #162.

Reply:
I've switched to `C_SIZEOF` which I hope will at least not introduce more chances for overflow than currently exist.